clang/test/dpct/warp.cu

```diff
@@ -337,6 +337,36 @@ __global__ void kernel35() {
   __shfl_xor_sync(0xFFFFFFFF, val, laneMask, WS);
 }
 
+__device__ void device36(int val, int laneMask, int WS) {
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1121:{{[0-9]+}}: Please initialize "val" which is used in the SYCL group function/algorithm.
+  // CHECK-NEXT: */
+  // CHECK-NEXT: dpct::permute_sub_group_by_xor(item_ct1.get_sub_group(), val, laneMask, WS);
+  __shfl_xor_sync(0xFFFFFFFF, val, laneMask, WS);
+}
+
+__global__ void kernel36() {
+  int val;
+  int laneMask;
+  int WS;
+  device36(val, laneMask, WS);
```

Comment:
please test if we need to add following two cases. 
 device36(val + 10, laneMak, WS)
int b=0;
 device36(val + b , laneMak, WS)

Reply:
done